### PR TITLE
WIP Dependency handling with venv

### PR DESCRIPTION
**Pull request by obrask1**
- Remove conda installation files
- Add venv environment installation with Make
- Update Readme
- Add Ruff linting tool

Wait until #32 is merged

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -1,10 +1,55 @@
 # pyproject.toml
 
+[build-system]
+requires = ["setuptools>=65.5.1", "wheel"]
+build-backend = "setuptools.build_meta"
+
+[project]
+# name and version fields are mandatory
+name = "fleet-learning"
+# To keep in sync with the project version specified in vccfl/__version__.py
```
Comment (jonasandersson10) [inline pyproject.toml:10]:
```suggestion
# To keep in sync with the project version specified in fleetlearning/__version__.py
```

**Review by jonasandersson10 — CHANGES_REQUESTED** [review]
Nice work Oscar. There are some things we need to fix before merging. Primarily, if we want to package the project as python package with this installation logic, I think we need to put everything into source a folder with the package name. This probably means we need to update all our imports. 

I've created another branch in which I've started to do some updates and add a python project build CI pipeline. My goal is to include the pytests as well there to make sure everything works. Once that's done we can merge that branch into this one :)

**Comment by obrask1** [comment]
> Nice work @obrask1. There are some things we need to fix before merging. Primarily, if we want to package the project as a python package with this installation logic, I think we need to move everything in `src` into a single folder with the package name. Right now I think we install multiple python-modules. Once we updated that, it probably means that we also need to update all our imports.
> 
> I've created another branch in which I've started to do some updates and add a python project build CI pipeline. My goal is to include the pytests as well there to make sure everything works. Once that's done we can merge that branch into this one :)

Great idea! Then I'll keep this PR as WIP until #32 is merged

**Comment by jonasandersson10** [comment]
@obrask1 All checks pass again now. Since we've merged my PR already, I think we are done with this or? What do you say?

**Comment by obrask1** [comment]
> @obrask1 All checks pass again now. Since we've merged my PR already, I think we are done with this or? What do you say?

agree!!

**Review by jonasandersson10 — APPROVED** [review]
🚀

**Pull request merged**